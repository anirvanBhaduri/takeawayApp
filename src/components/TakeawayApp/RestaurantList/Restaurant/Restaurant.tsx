import * as React from 'react';

/**
 * Restaurant Statuses.
 *
 * The following are the allowed status a restaurant can be in.
 *
 * @enum Status
 */
export const Status: {[index: string]: number} = {
    'open': 0,
    'closed': 1,
    'order ahead': 2,
};

/**
 * Favourite Props interface.
 *
 * Ensures a click handler is passed through to the
 * Favourite component.
 *
 * @interface FavouriteProps
 */
interface FavouriteProps {
    clickHandler: (e: any) => void;
    name: string;
    favourite?: boolean;
}

/**
 * Defines the format for a Restaurant object.
 *
 * @interface RestaurantProps
 *
 * @property string name
 * @property Status status
 * @property SortValues sortingValues
 */
export interface RestaurantProps extends FavouriteProps {
    status: string;
    sortingValues?: {
        [index: string]: number,
    };
}

/**
 * Favourite me message.
 *
 * @var string
 */
const favouriteMe = 'Favourite Me!';

/**
 * Un Favourite me message.
 *
 * @var string
 */
const unFavouriteMe = 'Un favourite me!';

/**
 * Render the favourite button.
 *
 * @param props
 *
 * @return JSX
 */
export const Favourite = (props: FavouriteProps) => {
    // For simplicity, we are going to make this a text based button
    // instead of an image.
    const favourite: string = props.favourite ? unFavouriteMe : favouriteMe;

    return (
        <button
            value={props.name}

            // We can also place a modifier here to change the appearance of the
            // button depending on the favourite state. To keep things simple,
            // we are going to omit the modifier here.
            className={'favourite'}

            onClick={props.clickHandler}
        >
            {favourite}
        </button>
    );
};

/**
 * Render a restaurant entry.
 *
 * @param props
 *
 * @return JSX
 */
export const Restaurant = (props: RestaurantProps) => {
    return (
        <div className={'restaurant'} >
            <div className={'restaurant__name'} >
                {props.name}
            </div>
            <div className={'restaurant__status'} >
                {props.status}
            </div>
            <Favourite
                name={props.name}
                favourite={props.favourite}
                clickHandler={props.clickHandler}
            />
        </div>
    );
};
