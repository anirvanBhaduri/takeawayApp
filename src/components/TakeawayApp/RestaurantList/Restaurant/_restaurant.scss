/// Restaurant component styles.
///
/// @author Anirvan Bhaduri
/// @since 10th Dec 2018

@import '../../../UI/_colour_constants.scss';
@import '../../../UI/size_constants';

// The Restaurant's styles
.restaurant {
  @include input-styles(30px, $app-body-width - 10);
  padding-top: 10px;
  border: 1px solid $takeaway-color;
  margin: 5px;
}

.restaurant__name {
  float: left;
  width: $app-body-width / 2;
  font-family: inherit;
}

.restaurant__status {
  float: left;
  font-style: italic;
  width: $app-body-width / 6;
  font-family: inherit;
}
