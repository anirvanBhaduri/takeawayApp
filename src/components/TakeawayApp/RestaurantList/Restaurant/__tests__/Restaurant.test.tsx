import * as React from 'react';
import {shallow} from "enzyme";

// SUT
import {Restaurant} from "../Restaurant";

/**
 * Restaurant component tests.
 *
 * @covers Restaurant
 */
describe('Restaurant Component Tests', () => {

    /**
     * Restaurant renders correctly.
     *
     * @covers Restaurant.render
     */
    it('should render correct snapshot of Restaurant Component', () => {
        const wrapper = shallow(
            <Restaurant
                status={'open'}
                clickHandler={() => {}}
                name={'My Restaurant.com!'}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
