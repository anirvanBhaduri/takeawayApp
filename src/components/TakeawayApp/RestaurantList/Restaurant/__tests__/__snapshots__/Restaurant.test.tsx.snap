// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant Component Tests should render correct snapshot of Restaurant Component 1`] = `
<div
  className="restaurant"
>
  <div
    className="restaurant__name"
  >
    <p>
      Name
    </p>
    My Restaurant.com!
  </div>
  <div
    className="restaurant__status"
  >
    <p>
      Status
    </p>
    open
  </div>
  <div
    className="restaurant__sort"
  >
    <p />
  </div>
  <Favourite
    clickHandler={[Function]}
    name="My Restaurant.com!"
  />
</div>
`;
